vertical-pod-autoscaler/pkg/updater/eviction/pods_eviction_restriction_test.go

```diff
@@ -521,6 +521,105 @@ func TestEvictAtLeastOne(t *testing.T) {
 	}
 }
 
+func TestEvictViaDeleteAtLeastOne(t *testing.T) {
+	replicas := int32(5)
+	livePods := 5
+	tolerance := 0.1
+
+	rc := apiv1.ReplicationController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rc",
+			Namespace: "default",
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "ReplicationController",
+		},
+		Spec: apiv1.ReplicationControllerSpec{
+			Replicas: &replicas,
+		},
+	}
+
+	cs := &apiv1.ContainerStatus{
+		RestartCount: 5,
+		LastTerminationState: apiv1.ContainerState{
+			Terminated: &apiv1.ContainerStateTerminated{
+				Reason: "OOMKilled",
+			},
+		},
+		State: apiv1.ContainerState{
+			Waiting: &apiv1.ContainerStateWaiting{
+				Reason: "CrashLoopBackOff",
+			},
+		},
+	}
+
+	pods := make([]*apiv1.Pod, livePods)
+	for i := range pods {
+		pods[i] = test.Pod().WithName(getTestPodName(i)).WithCreator(&rc.ObjectMeta, &rc.TypeMeta).WithContainerStatus(cs).Get()
+	}
+
+	factory, _ := getEvictionRestrictionFactory(&rc, nil, nil, 2, tolerance)
+	eviction := factory.NewPodsEvictionRestriction(pods, getBasicVpa())
+
+	for _, pod := range pods {
+		assert.True(t, eviction.CanEvict(pod))
+	}
+
+	for _, pod := range pods[:1] {
+		err := eviction.EvictViaDelete(pod, test.FakeEventRecorder())
+		assert.Nil(t, err, "Should evict with no error")
+	}
+	for _, pod := range pods[1:] {
+		err := eviction.EvictViaDelete(pod, test.FakeEventRecorder())
+		assert.Error(t, err, "Error expected")
```

Comment:
Why do we expect error here?

Reply:
This is effectively just a copy & paste of the unit test for the "normal" eviction. I found out that only one eviction is allowed per "replica controller" for every "tick" of the updater. So when multiple pods of the same replica set should get evicted it throws an error until the next tick.